[View in Colaboratory](https://colab.research.google.com/github/gr3ybr0w/cookbook/blob/master/Machine_Learning/Neural_Networks/TensorFlow/Keras/keras_101.ipynb)

In [0]:
import tensorflow as tf

Load and prepare the MNIST dataset. Convert the samples from integers to floating-point numbers:

In [3]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

11493376/11490434 [==============================] - 0s 0us/step


Build the tf.keras model by stacking layers. Select an optimizer and loss function used for training:

In [0]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(),
  tf.keras.layers.Dense(512, activation=tf.nn.relu),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation=tf.nn.softmax)
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

Train and evaluate model:

In [5]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test, y_test)

Epoch 1/5
60000/60000 [==============================] - 11s 176us/step - loss: 0.2180 - acc: 0.9353
Epoch 2/5
60000/60000 [==============================] - 9s 158us/step - loss: 0.0983 - acc: 0.9698
Epoch 3/5
60000/60000 [==============================] - 9s 156us/step - loss: 0.0688 - acc: 0.9789
Epoch 4/5
60000/60000 [==============================] - 9s 157us/step - loss: 0.0541 - acc: 0.9830
Epoch 5/5
10000/10000 [==============================] - 1s 53us/step


[0.06581762434233679, 0.981]

You’ve now trained an image classifier with ~98% accuracy on this dataset.